container-search/src/main/java/com/yahoo/search/yql/MinimalQueryInserter.java

```diff
@@ -35,44 +40,61 @@
 
     private static final CompoundName MAX_HITS = new CompoundName("maxHits");
     private static final CompoundName MAX_OFFSET = new CompoundName("maxOffset");
+    private static Logger log = Logger.getLogger(MinimalQueryInserter.class.getName());
 
-    @Override
-    public Result search(Query query, Execution execution) {
-        if (query.properties().get(YQL) == null) return execution.search(query);
+    @Inject
+    public MinimalQueryInserter(Linguistics linguistics) {
+        warmup(linguistics);
+    }
+    MinimalQueryInserter() {
+        this(new SimpleLinguistics());
+    }
+    static boolean warmup() {
+        return warmup(new SimpleLinguistics());
+    }
+    static boolean warmup(Linguistics linguistics) {
+        Query query = new Query("search/?yql=select%20*%20from%20sources%20where%20title%20contains%20'xyz';");
+        Result result = insertQuery(query, new ParserEnvironment().setLinguistics(linguistics));
+        if (result != null) {
+            log.warning("Something fishy. Reult = " + result.toString());
```

Comment:
"Result". But this message is in itself quite fishy. Could we make the error conditions here more crisp?

Reply:
Crisper..